README.md

```diff
@@ -14,73 +14,29 @@ def deps do
 
-If we instead want to have sub-keys `:b` and `:c` under the top-level `:a` key, we could do so like this:
+Data flows through SystemRegistry in two phases. First, process data is stored in a seperate fragment labeled by the caller pid and only contains the applied transactions of the caller. Second, the local pid fragment is then applied to the global state through a `SystemRegistry.Processor`.
 
-```elixir
-{:ok, { %{a: %{b: 1}}, %{} }} = SystemRegistry.update([:a, :b], 1)
-{:ok, { %{a: %{b: 1, c: 2}}, %{a: %{b: 1}} }} = SystemRegistry.update([:a, :c], 2)
-```
-
-At any time, you can call `match/1` to fetch the current value for a [`match_spec`](https://hexdocs.pm/elixir/Registry.html#match/3) in the registry.
-
-```elixir
-{:ok, {%{a: 1}, %{}}} = SystemRegistry.update([:a], 1)
-%{a: 1} = SystemRegistry.match(%{a: :_})
-%{} = SystemRegistry.match(%{b: :_})
-```
-
-Calling `delete/1` will return the current state and recursively trim the tree if intermediate nodes no longer have a value set.
-
-```elixir
-{:ok, {%{a: 1}, %{}}} = SystemRegistry.update([:a], 1)
-{:ok, %{}} = SystemRegistry.delete([:a])
-
-{:ok, {%{a: %{b: %{c: 1}}}, %{}}} = SystemRegistry.update([:a, :b, :c], 1)
-{:ok, %{}} = SystemRegistry.delete([:a, :b, :c])
-```
-
-SystemRegistry operates on a tree of nodes represented as nested maps, so if the value assigned to a scope is a Map, it is recursively expanded into scopes.
-
-```elixir
-{:ok, { %{a: %{b: 1}}, %{} }} = SystemRegistry.update([:a], %{b: 1})
-```
-
-#### Transactions
-
-Transactions let you compose `update` and `delete` functions using `transaction` and `commit` so they are executed atomically. Under the hood, `update/3` and `delete/2` pass a transaction through the pipeline and result in an atomic merged `update` and/or `delete` operation:
-
-```elixir
-{:ok, {%{a: 1, b: 2}, %{}}} =
-  SystemRegistry.transaction
-  |> SystemRegistry.update([:a], 1)
-  |> SystemRegistry.update([:b], 2)
-  |> SystemRegistry.commit
-```
-
-#### Processors
-
-Processors are workers that can perform operations based on transactions. SystemRegistry ships with a default `State` processor. Processors implement the `Processor` behaviour and have the capability of being consulted during transaction validation and notified of committed transactions.
+Processors are workers that implement the `SystemRegistry.Processor` behaviour and are the only means of moving data from local fragments to the global state. Processors implement two callbacks methods: `handle_validate/2` and `handle_commit/2`. A transaction can only be committed if all processors return `:ok` during the validation sequence. Processor validation errors are accumulated and returned in the case of an unsuccessful commit. SystemRegistry automaticlaly starts two processors for state and config.
```

Comment:
* "implement two callbacks methods" - callbacks should probably be "callback"
* "automatically" later in the paragraph